addons/hr_expense/wizard/hr_expense_split.py

```diff
@@ -57,7 +57,7 @@ def _onchange_product_id(self):
         In case we switch to the product without taxes defined on it, taxes should be removed.
         Computed method won't be good for this purpose, as we don't want to recompute and reset taxes in case they are removed on purpose during splitting.
         """
-        self.tax_ids = self.tax_ids if self.product_has_tax and self.tax_ids else self.product_id.supplier_taxes_id.filtered(lambda tax: tax.company_id == self.company_id)
+        self.tax_ids = self.product_id.supplier_taxes_id.filtered(lambda tax: tax.company_id == self.company_id)
```

Comment:
This is a functional change that clearly contradicts the behavior commented right above.
@sofiagvaladze what do you think of this?

Reply:
Indeed, 

I will answer a bit long answer, as user case helps to see why it was done so, originally

The wizard was done for the following user case in mind - Let's say employee went out with client, and they had meal (100 euro - taxed 10%) and drinks (50 euro - taxed 20%). And they are differently taxed. Employee submitted expense (maybe product - general expense or something) with 150 euro and adds 10% and 20% taxes on it. 

It's not really accurate. So we would like to split it. 

While splitting we would have 2 lines

150 euro | general expense| taxes 10% 20%

150 euro | general expense | taxes 10% 20%

Now we adjust amount and taxes

100 euro | general expense| taxes 10%

50 euro | general expense | taxes  20%

Looks better, but to be even more accurate I want to change the product as well, let's say I already have product alcohol (with 30% on it), and I change the second line. I would not like to overwrite taxes in this case as taxes come from recipe and I already set it. 

So, unless the behavior has changed, it's not a bug :)

Reply:
@Laura-labe So... Did the intended behavior change? :smile:

Reply:
We had discussed it a couple of months ago and Laura agreed with this task, see chatter of  https://www.odoo.com/web#id=3226278&cids=1&menu_id=4720&action=333&active_id=967&model=project.task&view_type=form
But I guess it's worth having a second look :eyes:

Reply:
Honestly I don’t disagree with you but I don’t totally agree either. In my opinion, the tax repartition configuration happens within the split wizard and not beforehand. The user starts by creating one expense -> he realizes that different taxes intervene -> he clicks on split expense -> He wants to be accurate so he start by deliberately changing the product (before changing the taxes). Because splitting an expense is basically like creating new expenses, the default vendor tax is expected to be applied when setting a product. 
So, I would indeed change the intended behavior :)